### ISSUE TITLE
Condition function never times out on command error

### ISSUE THREAD

**Issue by asfaltboy** [randy3k/UnitTesting#55]
For example, see the failing test case divmain/GitSavvy@63d9d78

**Comment by randy3k** [comment]
Thanks for the work. Unfortunately, I won't have time to look it up until next weekend.

**Comment by randy3k** [comment]
@asfaltboy 
Could you elborate more about the failing test case? It seems that it is working expectedly (failed).

https://travis-ci.org/divmain/GitSavvy/jobs/246551037

**Comment by asfaltboy** [comment]
Thanks for the fix in randy3k/UnitTesting@6990ade @randy3k . Yes I've seen that it worked well in travis, but for some reason this was "hanging" indefinitely when running "UnitTesting: " command in sublime (I've attached a screenshot in randy3k/UnitTesting#56).

<details>
  <summary>Now, after the fix in UnitTesting, it runs as expected in Sublime</summary>

```python
test_amend_commit (test_core_commands.test_gs_commit.TestAmend)
Test amending a recently added commit ... ok
test_amend_commit (test_core_commands.test_gs_commit.TestAmendFirstCommit)
Test amending the first commit ... Traceback (most recent call last):
  File "/Applications/Sublime Text.app/Contents/MacOS/sublime_plugin.py", line 818, in run_
    return self.run(edit)
  File "/Users/pavel/Library/Application Support/Sublime Text 3/Packages/GitSavvy/core/commands/commit.py", line 116, in run
    "HEAD^" if option_amend else None
  File "/Users/pavel/Library/Application Support/Sublime Text 3/Packages/GitSavvy/core/git_command.py", line 153, in git
    command_str, stdout, stderr
  File "/Users/pavel/Library/Application Support/Sublime Text 3/Packages/GitSavvy/core/git_command.py", line 112, in raise_error
    raise GitSavvyError(msg)
GitSavvy.core.exceptions.GitSavvyError: `/usr/bin/git diff --no-color --cached HEAD^` failed with following output:

fatal: ambiguous argument 'HEAD^': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

Condition timeout, continue anyway.
FAIL
test_add_and_commit_file (test_core_commands.test_gs_commit.TestCommit)
Test adding a new file and committing ... ok

======================================================================
FAIL: test_amend_commit (test_core_commands.test_gs_commit.TestAmendFirstCommit)
Test amending the first commit
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/pavel/Library/Application Support/Sublime Text 3/Packages/UnitTesting/unittesting/core/st3/case.py", line 13, in _executeTestPart
    for x in deferred:
  File "/Users/pavel/Library/Application Support/Sublime Text 3/Packages/GitSavvy/tests/test_core_commands/test_gs_commit.py", line 119, in test_amend_commit
    yield from self.assert_commit_view_rendered(first_line='Add foo')
  File "/Users/pavel/Library/Application Support/Sublime Text 3/Packages/GitSavvy/tests/test_core_commands/test_gs_commit.py", line 26, in assert_commit_view_rendered
    self.assertEqual(first_rows, [first_line or '', ''])
AssertionError: Lists differ: ['', ''] != ['Add foo', '']

First differing element 0:

Add foo

- ['', '']
+ ['Add foo', '']

----------------------------------------------------------------------
Ran 3 tests in 12.140s

FAILED (failures=1)

UnitTesting: Done.

```
</details>

**Comment by randy3k** [comment]
:+1 for `<details>` tag

doesn't know this feature at all.
